lib/rouge/lexers/kotlin.rb

```diff
@@ -48,6 +50,8 @@ class Kotlin < RegexLexer
           groups Keyword, Text
           push :function
         end
+        rule %r'(#{upper}#{name_backtick})(?=\s*[({])', Name::Class
```

Comment:
It's probably tempting fate to start playing around with other rules (who knows what you might break!) but does this eliminate the need for certain rules? If this identifies classes, for example, do you need these rules:

https://github.com/rouge-ruby/rouge/blob/7b876784f61444da6d3a9a8ada0a2ace8b13621a/lib/rouge/lexers/kotlin.rb#L33-L39

Reply:
Will see what happens without them, it may be that as you say they are covered in parts by a combination of other rules now.

Reply:
I have checked and yes you need these rules. These catch classes/types for properties or arguments to functions. They catch the occurrence of it after the colon.

```kotlin
val height: Int
live: Array<Array<Boolean>>
```

The new rule for class catches constructor calls, where there are brackets.

Reply:
Detecting classes can be simplified down much further, resulting in these rules not being needed